geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,52 +238,111 @@ object IngestCommand extends LazyLogging {
 
     var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
 
-    if (converter == null && inputs.nonEmpty) {
-      // ingesting from stdin without specifying a converter is not supported
-      if (inputs == Inputs.StdInInputs) {
-        throw new ParameterException("Cannot infer types from stdin - please specify a converter/sft or ingest from a file")
-      }
+    // if there is no converter passed in
+    if (converter == null && inputs.paths.nonEmpty) {
+      var open: () => InputStream = null
+      var file: FileHandle = null
+      var opened = ListBuffer.empty[CloseableIterator[InputStream]]
+
+      // try to infer the schema from stdin
+      if (inputs.paths == Inputs.StdInInputs) {
+        Command.user.info("No converter defined - will attempt to detect schema from stdin")
+
+        // Mark the beginning of the stdin stream so we can return to it later
+        System.in.mark(8192)
```

Comment:
seems like we shouldn't need to do this, and it's actually dangerous b/c according to the javadocs if more than 8192 bytes are read then it might not support reset()-ing back to the start.